test/api_cs_protocol_verification_test.rb

```diff
@@ -79,13 +79,21 @@ def test_fixtures
                end
 
         if code.to_s[0] == '2'
-          assert !@api.send(data['method'], *args).nil?
+          assert(!call_api(data['method'], args).nil?)
         else
-          assert_raises(MatrixSdk::MatrixRequestError.class_by_code(code)) { @api.send(data['method'], *args) }
+          assert_raises(MatrixSdk::MatrixRequestError.class_by_code(code)) { call_api(data['method'], args) }
         end
 
         @http.unstub(:request)
       end
     end
   end
+
+  def call_api(method, args)
+    if args.size.positive? && args.last.is_a?(Hash)
+      @api.send(method, *args[0..-2], **args.last)
```

Comment:
note: This needs to be reworked a bit, since this naive solution will cause `Passing the keyword argument as the last hash parameter is deprecated`, but I'll tackle this in follow-up.